frontend/src/metabase/dashboard/utils.ts

```diff
@@ -143,19 +143,21 @@ export function showVirtualDashCardInfoText(
   }
 }
 
-export function getAllDashboardCards(dashboard: Dashboard) {
+export function getAllDashboardCards(dashboard: Dashboard | null) {
+  if (!dashboard) {
+    return [];
+  }
+
   const results = [];
-  if (dashboard) {
-    for (const dashcard of dashboard.dashcards) {
-      const cards = [dashcard.card].concat((dashcard as any).series || []);
-      results.push(...cards.map(card => ({ card, dashcard })));
-    }
+  for (const dashcard of dashboard.dashcards) {
+    const cards = [dashcard.card].concat((dashcard as any).series || []);
+    results.push(...cards.map(card => ({ card, dashcard })));
   }
   return results;
 }
 
 export function getCurrentTabDashboardCards(
-  dashboard: Dashboard,
+  dashboard: Dashboard | null,
```

Comment:
wat? really we can try to get current tab dsahboard cards without the dashboard? looks like a wong type or usage

Reply:
I found this super weird as well. In this and `getAllDashboardCards`, the dashboard being passed in is sourced from `getDashboardComplete` which returns a `Dashboard | null`. So I can either `checkNotNull` that, or I can do this

Reply:
we shouldn't allow passing null on this level

Reply:
cool, I'll remove this, that makes me much happier